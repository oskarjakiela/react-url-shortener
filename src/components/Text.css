.Shorty-Text {
  font-family: 'Roboto', sans-serif;
  font-weight: 300;
  font-size: 1rem;
}

.Shorty-Text.Shorty-Text--primary {
  color: #555;
}

.Shorty-Text.Shorty-Text--secondary {
  color: #aaa;
}

.Shorty-Text.Shorty-Text--unimportant {
  color: #ccc;
}
