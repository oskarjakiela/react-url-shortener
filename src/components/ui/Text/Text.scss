.Shorty-Text {
  font-family: 'Roboto', sans-serif;
  font-weight: 300;
  font-size: 1rem;

  &--primary {
    color: #555;
  }

  &--secondary {
    color: #aaa;
  }

  &--unimportant {
    color: #ccc;
  }
}
