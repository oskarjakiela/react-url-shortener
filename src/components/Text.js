import classnames from 'classnames';
import PropTypes from 'prop-types';
import { all, equals, head } from 'ramda';
import React from 'react';

import './Text.css';


const isPrimary = (value) => {
  if (head(value)) {
    return true;
  }

  return all(equals(false), value)
}

const Text = ({
  children,
  primary,
  secondary,
  unimportant,
}) => (
  <span className={classnames('Text', {
    'Text--primary': isPrimary([primary, secondary, unimportant]),
    'Text--secondary': secondary,
    'Text--unimportant': unimportant,
  })}>{children}</span>
);

Text.defaultProps = {
  primary: false,
  secondary: false,
  unimportant: false,
};

Text.propTypes = {
  children: PropTypes.node.isRequired,
  primiary: PropTypes.bool,
  secondary: PropTypes.bool,
  unimportant: PropTypes.bool,
};

export default Text;
