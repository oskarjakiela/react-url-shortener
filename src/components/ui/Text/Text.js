import classnames from 'classnames';
import PropTypes from 'prop-types';
import { all, equals, head } from 'ramda';
import React from 'react';

import './Text.css';


const displayName = 'Shorty-Text';

const isPrimary = (value) => {
  if (head(value)) {
    return true;
  }

  return all(equals(false), value)
}

const Text = ({
  children,
  className,
  primary,
  secondary,
  unimportant,
  ...props,
}) => (
  <span
    className={classnames(displayName, {
      [`${displayName}--primary`]: isPrimary([primary, secondary, unimportant]),
      [`${displayName}--secondary`]: secondary,
      [`${displayName}--unimportant`]: unimportant,
    }, className)}
    {...props}
  >
    {children}
  </span>
);

Text.displayName = displayName;

Text.defaultProps = {
  primary: false,
  secondary: false,
  unimportant: false,
};

Text.propTypes = {
  children: PropTypes.node.isRequired,
  primiary: PropTypes.bool,
  secondary: PropTypes.bool,
  unimportant: PropTypes.bool,
};

export default Text;
