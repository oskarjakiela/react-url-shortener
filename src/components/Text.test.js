import React from 'react';
import renderer from 'react-test-renderer';

import Text from './Text';


it('<Text /> renders correctly', () => {
  const tree = renderer
    .create(<Text>Lorem ipsum</Text>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('<Text primary /> renders correctly', () => {
  const tree = renderer
    .create(<Text primary>Lorem ipsum</Text>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('<Text secondary /> renders correctly', () => {
  const tree = renderer
    .create(<Text secondary>Lorem ipsum</Text>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('<Text unimportant /> renders correctly', () => {
  const tree = renderer
    .create(<Text unimportant>Lorem ipsum</Text>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});


