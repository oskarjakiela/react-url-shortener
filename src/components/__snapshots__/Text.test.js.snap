// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders correctly 1`] = `
<span
  className="Shorty-Text Shorty-Text--primary"
>
  Lorem ipsum
</span>
`;

exports[`<Text className="foo-bar" /> renders correctly 1`] = `
<span
  className="Shorty-Text Shorty-Text--primary foo-bar"
>
  Lorem ipsum
</span>
`;

exports[`<Text primary /> renders correctly 1`] = `
<span
  className="Shorty-Text Shorty-Text--primary"
>
  Lorem ipsum
</span>
`;

exports[`<Text secondary /> renders correctly 1`] = `
<span
  className="Shorty-Text Shorty-Text--secondary"
>
  Lorem ipsum
</span>
`;

exports[`<Text title="foo-bar" disabled /> renders correctly 1`] = `
<span
  className="Shorty-Text Shorty-Text--primary"
  disabled={true}
  title="foo-bar"
>
  Lorem ipsum
</span>
`;

exports[`<Text unimportant /> renders correctly 1`] = `
<span
  className="Shorty-Text Shorty-Text--unimportant"
>
  Lorem ipsum
</span>
`;
